### PR TITLE
Disallow invocation of extension methods off of types as query operators.

### PR DESCRIPTION
**Pull request by AlekseyTs**
Fixes some of the scenarios for https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_workitems?id=204561&_a=edit.

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review Update 3 fix.

CC @MattGertz, @jaredpar For ask mode approval.

**Comment by MattGertz** [comment]
Meets the bar; approved pending CRs.

File: src/Compilers/CSharp/Test/Semantic/Semantics/QueryTests.cs [hunk 1]
```diff
@@ -2094,5 +2094,152 @@ where x.ToString() == y.ToString()
...
+        [WorkItem(204561, "https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_workitems?id=204561&_a=edit")]
+        public void Bug204561_01()
+        {
+            string sourceCode =
+@"
+class C
+{
+    public static void Main()
+    {
+        var xxx = from a in Test select a + 1;
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Semantic/Semantics/QueryTests.cs]: Use a different variable name please.

**Comment by jcouv** [comment]
LGTM with other arbitrary variables names.

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review Update 3 fix, I need a second sign-off.

**Comment by gafter** [comment]
LGTM

**Pull request merged**